pkg/controllermanager/deployer/deployer.go

```diff
@@ -386,15 +387,20 @@ func (deployer *Deployer) populateBasesAndLocalizations(sub *appsapi.Subscriptio
 			baseTemplate.Labels[known.ClusterNameLabel] = ns.Labels[known.ClusterNameLabel]
 		}
 
-		basesToBeDeleted.Delete(klog.KObj(baseTemplate).String())
-		base, err := deployer.syncBase(sub, baseTemplate)
+		newBaseName := fmt.Sprintf("%s-%s", sub.Name, utilrand.String(5))
```

Comment:
wait.....

If we are using random string in base names, we CAN NOT use listers to GET a specific base object, since the names are always changing.